components/component.go

```diff
@@ -97,6 +97,73 @@ func (c *Component) ConfigComponentLogger(logger logr.Logger, component string,
 	return logger.WithName("DSC.Components." + component)
 }
 
+func getJobName(job map[any]any) (string, bool) {
```

Comment:
have not touched the detail of this PR, but should `component.go` change get into ODH first?

Reply:
> have not touched the detail of this PR, but should `component.go` change get into ODH first?

100%

It's a branch to test it first.

Reply:
I was thinking that in downstream they anyway should come as one patch (upstream changes do not break anything there, no problem), but looks like having the code without prometheus changes even if does not work as expected, does not break existing setup.

Reply:
yeah, this is similar situation to MR's PR. 
we can use this one to test first and backport to ODH as long as we have the logic applied to both upstream and downstream.

Reply:
I'll split it for the real submission, it's fine.